solrmonitor/collectionstate.go

```diff
@@ -14,20 +14,50 @@
 
 package solrmonitor
 
+import "fmt"
+
 type CollectionState struct {
-	Shards            map[string]ShardState `json:"shards"`            // map from shard name to shard state
-	ReplicationFactor string                `json:"replicationFactor"` // e.g. "1" (yes, these are strings, not numbers)
-	Router            Router                `json:"router"`            // e.g. {"name":"compositeId"}
-	MaxShardsPerNode  string                `json:"maxShardsPerNode"`  // e.g. "1" (yes, these are strings, not numbers)
-	AutoAddReplicas   string                `json:"autoAddReplicas"`   // e.g. "false" (yes, these are strings, not bools)
+	Shards            map[string]*ShardState `json:"shards"`            // map from shard name to shard state
+	ReplicationFactor string                 `json:"replicationFactor"` // e.g. "1" (yes, these are strings, not numbers)
+	Router            Router                 `json:"router"`            // e.g. {"name":"compositeId"}
+	MaxShardsPerNode  string                 `json:"maxShardsPerNode"`  // e.g. "1" (yes, these are strings, not numbers)
+	AutoAddReplicas   string                 `json:"autoAddReplicas"`   // e.g. "false" (yes, these are strings, not bools)
 
 	// These fields are synthetic. They ARE present in COLLECTIONSTATUS response in
 	// solr collection API, but they are NOT in state.json docs in Zookeeper.
 
-	ConfigName    string `json:"configName,omitempty"`   // the name of the node in solr/configs (in ZK) that this collection uses
-	ZkNodeVersion int32  `json:"znodeVersion,omitempty"` // the ZK node version this state snapshot represents
+	ConfigName      string `json:"configName,omitempty"`   // the name of the node in solr/configs (in ZK) that this collection uses
+	ZkNodeVersion   int32  `json:"znodeVersion,omitempty"` // the ZK node version this state snapshot represents
+	PerReplicaState string `json:"perReplicaState"`        // whether collection keeps state for each replica separately
```

Comment:
This is a string?  Not like, a boolean?

Reply:
Not sure why solr has this as string
```
    "tlogReplicas": "0",
        "perReplicaState": "true"
```

Reply:
That is unfortunate. :/. I thought we're the ones who implemented PRS?

Reply:
yeah. not sure why Ishan/Noble used this as string. Maybe I will ask them!